### ISSUE TITLE
UnusedParam false-positive when other issues before

### ISSUE THREAD

**Issue by homersimpsons** [vimeo/psalm#11186]
On the following code, psalm reports an UnusedParam for `$userid` and `$error`:

https://github.com/BOINC/boinc/blob/275a2a42b7004e84cb4245161211ce0a99bec7cf/html/inc/pm.inc#L113-L209

I do not really understand why, because the `$userid` param is used on:

https://github.com/BOINC/boinc/blob/275a2a42b7004e84cb4245161211ce0a99bec7cf/html/inc/pm.inc#L145

If you want to run psalm in the `html/` directory above, it will take ~40 minutes. By trying to reduce the code I came to (https://psalm.dev/r/0a1e623634):

```php
function pm_form_page($replyto, $userid) {
    if ($replyto) {
        $userid = 'a';
        User::lookup_id($userid);
    } elseif ($userid) {
        $user = User::lookup_id($userid);
    }
}
```

which highlight the same issue. But I do not know if this is the same root cause. Note that the issue disappears if I add the following above:

```php
class User {
    public static function lookup_id($userid) {}
}
```

I searched if this issue was already reported but it looks like no. There are other issues about UnusedParam false-positive, but they seem to be about other cases.
